test/test_sam_annotator/test_cli.py

```diff
@@ -83,9 +85,47 @@ def test_precompute_embeddings(self):
             ais_path = os.path.join(emb_path3, f"image-{i}.zarr", "is_state.h5")
             self.assertTrue(os.path.exists(ais_path))
 
+    @pytest.mark.skipif(platform.system() == "Windows", reason="CLI test is not working on windows.")
     def test_automatic_segmentation(self):
         self._test_command("micro_sam.automatic_segmentation")
 
+        # Create 1 image as testdata.
+        im_path = os.path.join(self.tmp_folder, "image.tif")
+        image_data = binary_blobs(512).astype("uint8") * 255
+        imageio.imwrite(im_path, image_data)
+
+        out_path = "output.tif"
+
+        # Test AMG with default model in default mode.
+        run(
+            ["micro_sam.automatic_segmentation", "-i", im_path, "-o",
+             out_path, "-m", self.default_model_type]
+        )
```

Comment:
For these tests: 
- We should check that the output path exists after each `run` and then remove it (so that the checks are reliable for the next function).
- Update the kwargs for all that use `amg` to speed up the tests (amg with default arguments are very slow), e,g, set `points_per_side=4`.

Reply:
> We should check that the output path exists after each run and then remove it (so that the checks are reliable for the next function).

Okay, yes. This one's easy. I'll take care of it.

> Update the kwargs for all that use amg to speed up the tests (amg with default arguments are very slow), e,g, set points_per_side=4.

Merging this in our current argument parsing setup via CLI is a bit tricky (because some arguments, eg. `points_per_side` are expected in `__init__` and others, eg. `stability_score_thresh`, are expected in the `generate` method, which in the automatic segmentation functionality are at different levels). Maybe let's look at it together to figure out an optimal solution.

Reply:
Now that I gave it a second look, it is not as complicated (the internal logic devised at the moment is quite detailed), but on the CLI level it isn't.

I'll add the changes now in the latest commit. (thanks for mentioning this)